src/workarounds/forwarddiff_rules.jl

```diff
@@ -235,10 +234,20 @@ function self_consistent_field(basis_dual::PlaneWaveBasis{T};
     end
     ρ_out = map((ρi, δρi...) -> DT(ρi, δρi), scfres.ρ, δρ...)
 
-    # TODO Compute eigenvalue response (return dual eigenvalues and dual εF)
+    ## Build consistent Hamiltonian and band energies
+    energies_out, ham_out = energy_hamiltonian(basis_dual, ψ_out, occupation_dual;
+                                               ρ=ρ_out, eigenvalues=eigenvalues_dual,
+                                               εF=εF_dual)
+
+    eigenvalues_out = map(scfres.eigenvalues, ψ_out, ham_out * ψ_out) do εk, ψk, Hψk
+        # Primal and 1st order change due to updated orbitals
+        # The second term accounts for the loss of orthogonality in the
+        # orbitals due to the perturbation (We don't enforce δψnk ⟂ ψnk)
+        eigvals(Hermitian(ψk'Hψk + Diagonal(εk) * (I - ψk'ψk)))
```

Comment:
That looks convoluted, are you sure this is correct and if yes can you expand on the rationale? Otherwise you can do `eigvals(psi' H psi, psi' psi)`. In any case this should be done at the level of the response solver, since that's where we decide on conventions. We can either decide <dpsink, psink> = 0, depsnk = <psink, dH psink>, or depsnk=0 and <dpsink, psink> != 0. I'd favor the first solution; in which case the response solvers should output depsnk as well as dpsi

Reply:
Also here you're taking the Hermitian of a non-hermitian matrix, which is tricky

Reply:
cc @gkemlin

Reply:
> `eigvals(psi' H psi, psi' psi)`

You can, but only using GenericLinearAlgebra, which I'd like to avoid. The answer is the same, though.

> should be done at the level of the response solver,

Yes, I agree, but the whole point of our response setup is exactly to have <dpsink, psink> != 0, right? Or do you mean to "undo" that convention before returning the response result in this function.

Also why is the matrix not Hermitian? I don't see that (but it might be too early).

Reply:
Yes indeed, the whole point of the response solver is to make another gauge choice than <dpsink, psink> = 0 for metals

Reply:
> Also why is the matrix not Hermitian? I don't see that (but it might be too early).

It's not unless the psik are eigenvectors (but it might also be to first order, I haven't checked)

OK, you're right, I think I was confusing with the dfnk, sorry. In that case we can still return the deps_nk from the response solver, with the understanding that psink+dpsink might not be an eigenvector. I would be more comfortable if we just computed the deps_nk from <psink, dH psink> than from this relatively expensive and nontrivial formula. (note dH has to be the total hamiltonian, not the external one)

Reply:
I think that we also defined deps_nk as <psink, dH psink> back in Aachen, which also comes from the first order perturbation no ?

Reply:
@gkemlin Yes, but that's not what you get with finite differences unfortunately. My intention was to match the finite differences result (see the updated test) and for that I had to update the Hamiltonian once more.

Reply:
Ok scratch that. I went the other way to use the result from solve_Omega_plus_K